lib/modules/datasource/aws-eks-addon/index.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  type AddonInfo,
+  type AddonVersionInfo,
+  type Compatibility,
+  DescribeAddonVersionsCommand,
+  type DescribeAddonVersionsCommandInput,
+  type DescribeAddonVersionsCommandOutput,
+  EKSClient,
+} from '@aws-sdk/client-eks';
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+// import * as awsEksAddonVersioning from '../../versioning/aws-eks-addon';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { EksAddonsFilter } from './schema';
+
+export class AwsEKSAddonDataSource extends Datasource {
+  static readonly id = 'aws-eks-addon';
+
+  // override readonly defaultVersioning = awsEksAddonVersioning.id;
+  override readonly caching = true;
+  private readonly clients: Record<string, EKSClient> = {};
+
+  override readonly defaultConfig: Record<string, unknown> | undefined = {
+    commitMessageTopic: '{{datasource}}',
+    commitMessageExtra: '{{currentVersion}} to {{{newVersion}}}',
+  };
+
+  constructor() {
+    super(AwsEKSAddonDataSource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${AwsEKSAddonDataSource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `getReleases:${packageName}`,
+  })
+  async getReleases({
+    packageName: serializedFilter,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const res = EksAddonsFilter.safeParse(serializedFilter);
+    if (!res.success) {
+      logger.error(
+        { err: res.error, serializedFilter },
+        'Error parsing eks-addons config.',
+      );
+      return null;
+    }
+
+    const filter = res.data;
+    const input: DescribeAddonVersionsCommandInput = {
+      kubernetesVersion: filter?.kubernetesVersion,
+      addonName: filter?.addonName,
+      maxResults: 1,
+    };
+
+    const cmd = new DescribeAddonVersionsCommand(input);
+    const response: DescribeAddonVersionsCommandOutput =
```

Comment:
```suggestion
    const response =
```
i think result tyoe can be inferred by typescript